examples/refrigerator-app/silabs/src/RefrigeratorManager.cpp

```diff
@@ -0,0 +1,207 @@
+}
+
+
+void RefrigeratorManager::RefAndTempCtrlAttributeChangeHandler(EndpointId endpointId, AttributeId attributeId, uint8_t * value, uint16_t size)
+{
+    switch (attributeId)
+    {
+        case RefAndTempAttr::CurrentMode::Id: {
+            int8_t currentMode = static_cast<int16_t>(*value);
+            mCurrentMode = currentMode;
+        }
+        break;
+
+        case RefAndTempAttr::StartUpMode::Id: {
+            int8_t startUpMode = static_cast<int16_t>(*value);
+            mStartUpMode = startUpMode;
+        }
+        break;
+
+        case RefAndTempAttr::OnMode::Id: {
+            int8_t onMode = static_cast<int16_t>(*value);
+            mOnMode = onMode;
+        }
+        break;
+
+        default: {
+            SILABS_LOG("Unhandled Refrigerator and Temprature attribute %x", attributeId);
+            return;
+        }
+        break;
+    }
+}
+
+void RefrigeratorManager::TempCtrlAttributeChangeHandler(EndpointId endpointId, AttributeId attributeId, uint8_t * value, uint16_t size)
+{
+    switch (attributeId)
+    {
+        case TempCtrlAttr::TemperatureSetpoint::Id: {
+            int16_t temperatureSetpoint = ConvertToPrintableTemp(*((int16_t *) value));
+            mTemperatureSetpoint = temperatureSetpoint;
+            TempCtrlAttr::TemperatureSetpoint::Set(endpointId, temperatureSetpoint);
+        }
+        break;
+
+        case TempCtrlAttr::MinTemperature::Id: {
+            int16_t minTemperature = ConvertToPrintableTemp(*((int16_t *) value));
+            mMinTemperature = minTemperature;
+            TempCtrlAttr::MinTemperature::Set(endpointId, minTemperature);
+        }
+        break;
+
+        case TempCtrlAttr::MaxTemperature::Id: {
+            int16_t maxTemperature = ConvertToPrintableTemp(*((int16_t *) value));
+            mMaxTemperature = maxTemperature;
+            TempCtrlAttr::MaxTemperature::Set(endpointId, maxTemperature);
+        }
+        break;
+
+        case TempCtrlAttr::SelectedTemperatureLevel::Id: {
+            int16_t selectedTemperatureLevel = ConvertToPrintableTemp(*((int16_t *) value));
+            mSelectedTemperatureLevel = selectedTemperatureLevel;
+            TempCtrlAttr::SelectedTemperatureLevel::Set(endpointId, selectedTemperatureLevel);
+        }
+        break;
+        
+        case TempCtrlAttr::Step::Id: {
+            int16_t step = ConvertToPrintableTemp(*((int16_t *) value));
+            TempCtrlAttr::MaxTemperature::Set(endpointId, step);
+        }
+        break;
+
+        default: {
+            SILABS_LOG("Unhandled Temprature controlled attribute %x", attributeId);
+            return;
+        }
+        break;
+    }
+
+    // AppTask::GetAppTask().UpdateRefUI();
+}
+
+void RefrigeratorManager::RefAlaramAttributeChangeHandler(EndpointId endpointId, AttributeId attributeId, uint8_t * value, uint16_t size)
+{
+    switch (attributeId)
+    {
+        case RefAlarmAttr::Mask::Id: {
+            int8_t mask = static_cast<uint32_t>(*value);
```

Comment:
Mismatch in the cast and assignment data types and may lead to data loss or unexpected behavior?

Reply:
updated